src/components/Common/AutoFocusInput.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <b-form-input
+    :ref="refName"
+    v-model="value"
+    v-bind="$attrs"
+    @blur="looseFocus"
+    data-auto-focus="true"
+  />
+</template>
+
+<script>
+export default {
+  name: 'AutoFocusInput',
+  props: {
+    name: {
+      type: String,
+      required: true
+    },
+    initialValue: {
+      type: String,
+      default: ''
+    }
+  },
+  computed: {
+    refName() {
+      return `auto-focus-input-${this.name}`
+    }
+  },
+  watch: {
+    value () {
+      this.$emit('input', this.value)
+    }
+  },
+  data () {
+    return {
+      value: this.initialValue,
+      isFocus: false
+    }
+  },
+  methods: {
+    // AutoFocusInput is only activated
+    enable () {
+      const tagType = document.activeElement.tagName
+
+      // If we are currently selecting another input
+      if (tagType === 'INPUT' && !document.activeElement.dataset.autoFocus) {
+        return false
+      }
+
+      return true
+    },
+    looseFocus() {
+      this.isFocus = false
+    },
+    listenKeypress () {
+      document.onkeypress = keyEvent => {
+        if (! this.enable()) {
+          return
+        }
+
+        if (this.isFocus) {
+          if (keyEvent.code === 'Enter') {
+            this.$emit('submit')
+          }
+
+          return;
+        }
+
+        this.isFocus = true;
+        this.$refs[this.refName].focus()
+
+        // Firefox does not follow the key event to the input but Chrome does
+        // so after the event has finished his propagation, we check the input
+        // value and on Firefox it will be empty so we put the first letter inside
+        // Using preventDefault() to manually input the first letter and prevent
+        // to have double input on Chrome doesn't work well because then the cursor
+        // is at the wrong place
+        setTimeout(() => {
```

Comment:
This whole thing looks very convoluted to me, or I don't understand what we're trying to do.

Reply:
As the comment say, I have a different behavior on Chrome and Firefox and I use this tricks to make it works in both.

An AutoFocusInput is an input which will be auto focused in the user start to type on his keyboard.  
For example on Github if you start to type without focusing any input then you will autofocus the Search Bar

Reply:
Ok, now I get the purpose of this whole component. Still... I'd say the responsibility of auto-focusing an input should be of the root component on the page, rather than a single component. Like, what if someone adds many `AutoFocusInput`s in the same page?

Reply:
They shouldn't do that :sweat_smile:

Reply:
Eventually I can add a guard to prevent developer to make this mistake

Reply:
Well, I think this is not good. A leaf component shouldn't have the responsibility of a scope greater than its own one. I'm still convinced that the auto-focus is a concern of the page, not the input element. To me, this component should not exist and the auto-focus logic should be moved to the page.